src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs

```diff
@@ -68,5 +68,33 @@ public T Get<T>(string id, DateTime date)
             operation.SetResult(command.Result);
             return operation.GetRevisionsFor<T>().FirstOrDefault();
         }
+        
+        public void ForceRevisionCreationFor<T>(T entity, ForceRevisionStrategy forceRevisionCreationStrategy = ForceRevisionStrategy.Before)
+        {
+            if (ReferenceEquals(entity, null))
+                throw new ArgumentNullException(nameof(entity));
+           
+            if (DocumentsByEntity.TryGetValue(entity, out DocumentInfo documentInfo) == false)
+            {
+                throw new InvalidOperationException("Cannot create a revision for the requested entity because it is not tracked by the session");
+            }
+
+            if (Session.IDsForCreatingForcedRevisions.ContainsKey(documentInfo.Id))
+            {
+                throw new InvalidOperationException("A request for creating a revision was already made for the document in the current session.");
+            }
+            
+            Session.IDsForCreatingForcedRevisions.Add(documentInfo.Id, forceRevisionCreationStrategy);
+        } 
+        
+        public void ForceRevisionCreationFor(string id, ForceRevisionStrategy forceRevisionCreationStrategy = ForceRevisionStrategy.Before)
+        {
+            if (Session.IDsForCreatingForcedRevisions.ContainsKey(id))
+            {
```

Comment:
Same

Reply:
Done. Now throwing only if a different strategy is requested in the same session.